server/src/main/java/org/apache/iotdb/db/engine/cache/TimeSeriesMetadataCache.java

```diff
@@ -60,6 +64,7 @@
 
   private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
+  private final Map<String, WeakReference<String>> devices = Collections.synchronizedMap(new WeakHashMap<>());
```

Comment:
this should be discussed. 
We may have too many devices and the memory is limited..

Reply:
Hi, @jixuan1989 . Thanks for your reply. Indeed, taking this into consideration, so WeakReference is used here, which is more GC friendly. Moreover, I think the previous method has the risk of deadlock and the memory usage is not small.

Reply:
Hi, @JackieTien97 . FYI, https://stackoverflow.com/a/134154/4719867